### PR TITLE
move all authentication logic to server

### LINKED ISSUE #126: Server-side auth
Move app authentication to server

### PR DESCRIPTION
**Pull request by R1ZZU**
Closes #126, #119, and #88

### CODE REVIEW

File: config/routes/index.coffee [hunk 1]
```diff
@@ -8,7 +9,15 @@ module.exports = (app) ->
 		if @isAuthenticated()
 			@cookies.set 'auth', @passport.user.role, signed: yes
 			yield next
+
+		else if cssqd_config.get('service:blind_guardian')
```
Comment (DaQuirm) [inline config/routes/index.coffee]: Mr. Volchenko, tear down those parentheses! :smile_cat:

File: public/game-master/app.coffee [hunk 2]
```diff
@@ -2,36 +2,12 @@ require 'normalize.css'
...
-require 'fetch'
 
 AppViewModel = require './view-models/app-viewmodel'
 AppView      = require './views/app-view'
-urlUtils     = require '../common/utils/url-utils'
 
-ensureSessionId = ->
-	sessionId = urlUtils.getParamValue urlUtils.getParamsString(location.search), 'id'
+if not window.sessionId
+	throw new Error 'Unrecognizable session if (URL parameter [id]). Application may fail to start'
```
Comment (DaQuirm) [inline public/game-master/app.coffee]: Resembles a log message from DAWN...
Some wounds never heal :crying_cat_face:

File: config/routes/index.coffee (hunk 1 above)
Comment (R1ZZU) [inline config/routes/index.coffee]: Copy-pasted from [here](https://github.com/DaQuirm/css-quickdraw-redux/pull/132/files/a197cd0f6b297aaa4f8356b4bf14e54f7f778f65#diff-1b8d53667fa02c72249bf8bbfc35b3a6L13) :disappointed:

Comment (DaQuirm) [inline config/routes/index.coffee]: But it's justified there because of the following `is`...

**Comment by DaQuirm** [comment]
Wow, looks very streamlined and straightforward! GJ :smile_cat:

**Comment by DaQuirm** [comment]
Code LGTM :coffee: 
Making gunslinger accept the new reality...

File: config/routes/private.coffee [hunk 3]
```diff
@@ -3,4 +3,16 @@
 Router = require 'koa-router'
 router = new Router
 
+router.get '/game', ->
+	if @passport.user.role is 'game_master'
```
Comment (DaQuirm) [inline config/routes/private.coffee]: Mmmm, didn't we decide to drop `user.role` in favor of per game session roles?

**Pull request merged**